### PR TITLE
Feature/fixed length pu

### PR DESCRIPTION
**Pull request by pl8787**
#251

### CODE REVIEW

File: matchzoo/preprocessor/process_units.py [hunk 1]
```diff
@@ -335,3 +335,55 @@ def transform(
             idx = self._term_index.get(key, 0)
             hashing[idx] = value
         return hashing
+
+
+class FixedLengthUnit(ProcessorUnit):
+    """Process unit to get the fixed length text."""
+
+    def __init__(self, text_length, pad_value=0, pad_mode='pre', truncat_mode='pre'):
```
Comment (faneshion) [inline matchzoo/preprocessor/process_units.py]: Why not use `truncate_mode`? Any criteria to  discard 'e'?

File: matchzoo/preprocessor/process_units.py [hunk 2]
```diff
@@ -335,3 +335,56 @@ def transform(
             idx = self._term_index.get(key, 0)
             hashing[idx] = value
         return hashing
+
+
+class FixedLengthUnit(ProcessorUnit):
+    """Process unit to get the fixed length text."""
+
+    def __init__(self, text_length, pad_value=0,
+                 pad_mode='pre', truncate_mode='pre'):
```
Comment (bwanglzu) [inline matchzoo/preprocessor/process_units.py]: Missing type hints

File: matchzoo/preprocessor/process_units.py [hunk 3]
```diff
@@ -335,3 +335,56 @@ def transform(
...
+        """
+        Class initialization.
+
+        :param text_length: fixed length of the text.
+        :param pad_value: if text length is smaller than :attr: `text_length`,
+            filling text with :attr: `pad_value`.
```
Comment (bwanglzu) [inline matchzoo/preprocessor/process_units.py]: Remove the space between:attt: and ‘

File: matchzoo/preprocessor/process_units.py [hunk 4]
```diff
@@ -335,3 +335,56 @@ def transform(
...
+        :param pad_mode: String, 'pre' or 'post':
```
Comment (bwanglzu) [inline matchzoo/preprocessor/process_units.py]: `pre` `post`

File: matchzoo/preprocessor/process_units.py [hunk 5]
```diff
@@ -335,3 +335,56 @@ def transform(
...
+        :param truncate_mode: String, 'pre' or 'post':
+            remove values from sequences larger than :attr: `text_length`,
+            either at the beginning or at the end of the sequences.
+        """
+        self._text_length = text_length
+        self._pad_value = pad_value
+        self._pad_mode = pad_mode
+        self._truncate_mode = truncate_mode
+
+    def transform(self, tokens: list) -> list:
```
Comment (bwanglzu) [inline matchzoo/preprocessor/process_units.py]: Return type is list or nparray?

File: matchzoo/preprocessor/process_units.py [hunk 6]
```diff
@@ -335,3 +335,56 @@ def transform(
             idx = self._term_index.get(key, 0)
             hashing[idx] = value
         return hashing
+
+
+class FixedLengthUnit(ProcessorUnit):
+    """Process unit to get the fixed length text."""
```
Comment (bwanglzu) [inline matchzoo/preprocessor/process_units.py]: I suggest to add example here to show the usage I.e

Example:
    >>>

**Pull request merged**